File: microbenchmarks/cpp/integer_boolean/big_integer/Negative.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef Negative_H
#define Negative_H

using namespace std;

class Negative {



    private:



        BInteger counter;
        BInteger value;

    public:

        Negative() {
            counter = (BInteger("0"));
            value = (BInteger("0"));
        }

        void simulate() {
            while((counter.less((BInteger("5000000")))).booleanValue()) {
                counter = counter.plus((BInteger("1")));
                value = value.negative();
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    Negative exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

File: microbenchmarks/java/integer_boolean/Java_Primitive_Integer/Negative.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class Negative {






    private BInteger counter;
    private BInteger value;

    public Negative() {
        counter = new BInteger(0);
        value = new BInteger(0);
    }

    public void simulate() {
        while((counter.less(new BInteger(5000000))).booleanValue()) {
            counter = counter.plus(new BInteger(1));
            value = value.negative();
        }
    }

    public static void main(String[] args) {
        Negative exec = new Negative();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}
